VS_Meadow_Extension/VS_Meadow_Extension.Shared/DeployProvider.cs

```diff
@@ -43,35 +47,72 @@ public DeployProvider(ConfiguredProject configuredProject)
 
         public async Task DeployAsync(CancellationToken cts, TextWriter outputPaneWriter)
         {
-            await DeployOutputLogger?.ConnectTextWriter(outputPaneWriter);
-            MeadowPackage.DebugOrDeployInProgress = false;
-
-            var generalProperties = await Properties.GetConfigurationGeneralPropertiesAsync();
-            var name = await generalProperties.Rule.GetPropertyValueAsync("AssemblyName");
-
-            //This is to avoid repeat deploys for multiple projects in the solution
-            if (name != "App")
+            if (cts.IsCancellationRequested)
             {
                 return;
             }
 
-            MeadowPackage.DebugOrDeployInProgress = true;
-
-            var projectDir = await generalProperties.Rule.GetPropertyValueAsync("ProjectDir");
-            var outputPath = Path.Combine(projectDir, await generalProperties.Rule.GetPropertyValueAsync("OutputPath"));
+            // Get the currently selected project
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            try
+            var dte = Package.GetGlobalService(typeof(DTE)) as DTE;
+            if (dte == null)
             {
-                await DeployAppAsync(Path.Combine(projectDir, outputPath), new OutputPaneWriter(outputPaneWriter), cts);
+                return;
             }
-            catch (Exception ex)
+
+            var solution = dte.Solution;
+            var startupProjects = solution.SolutionBuild.StartupProjects;
+            if (startupProjects == null)
             {
-                MeadowPackage.DebugOrDeployInProgress = false;
+                return;
+            }
 
-                DeployOutputLogger?.Log($"Deploy failed: {ex.Message}{Environment.NewLine}StackTrace:{Environment.NewLine}{ex.StackTrace}");
-                DeployOutputLogger?.Log("Reset Meadow and try again.");
+            MeadowPackage.DebugOrDeployInProgress = false;
 
-                throw ex;
+            foreach (string filename in (Array)startupProjects)
+            {
+                if (!filename.EndsWith(".csproj"))
+                {
+                    continue;
+                }
+                var csprojContent = File.ReadAllText(filename);
+                if (csprojContent.Contains("Sdk=\"Meadow.Sdk/1.1.0\""))

```

Comment:
Wow, VS automation doesn't have a cleaner way to determine project type?  You wouldn't think we were the only ones to ever need this capability.

Reply:
There's a project type GUID, it's just finding which COM object it's hidden in.